configure.ac

```diff
@@ -271,23 +259,86 @@ case $target in
 	  enable_pkey=yes
 	  ;;
      *s390*)
-	  CFLAGS="$CFLAGS -m31 -DNO_PKEY"
+	  CFLAGS="$CFLAGS -m31"
+		AC_DEFINE([NO_PKEY])
 	  ;;
      *ppc64* | *x86_64*)
-	  CFLAGS="$CFLAGS -DNO_PKEY"
+		AC_DEFINE([NO_PKEY])
 	  ;;
      *ppc* | i*86*)
-	  CFLAGS="$CFLAGS -m32 -DNO_PKEY"
+	  CFLAGS="$CFLAGS -m32"
+		AC_DEFINE([NO_PKEY])
 	  ;;
      *)
-	  CFLAGS="$CFLAGS -DNO_PKEY"
+		AC_DEFINE([NO_PKEY])
 	  ;;
 esac
 AM_CONDITIONAL([NO_PKEY], [test "x$enable_pkey" != "xyes"])
 
+dnl Check and save whether the platform is AIX
+dnl Using the pthread library requires the _THREAD_SAFE macro to be set
+build_aix=no
+case "${host_os}" in
+    aix*)
+	     build_aix=yes
+			 AC_DEFINE([_ALL_SOURCE])
+			 AC_DEFINE([_THREAD_SAFE])
+			 with_systemd=no
+			 with_libica=no
+			 enable_ccatok=yes
+			 enable_swtok=yes
+			 enable_tpmtok=no
+			 enable_icsftok=no
+			 enable_ep11tok=no
+			 echo "Disabling migration utilities on AIX: pkcscca pkcstok_migrate"
+			 enable_pkcscca=no
```

Comment:
You mention `pkcscca` is disabled due to co-dependency on CHIM. I don't quite understand what you mean by that.

I don't think that you should completely disable `pkcscca` for AIX.
`pkcscca` does different things. Most of them you don't need, such as command `v2objectsv3` or `oldrsakeys`, because there is no old version on AIX. However, `keys` is to re-encipher the CCA keys in case of an (offline) CCA master key change. I think that this IS required, even on AIX. 

If you disable `pkcscca` entirely, then you are left to concurrent CCA master key change via `pkcshsm_mkchage` only. Not sure if this is really what you want. Some users might prefer to do that in an offline way, not concurrently.

Reply:
Okay, I will revisit it alongside ICSF token support. But it will remain disabled for now.